cmd/controller/main.go

```diff
@@ -53,8 +61,19 @@ func main() {
 		auth.OIDCLabelSelector,
 		eventingtls.TrustBundleLabelSelector,
 		sinks.JobSinkJobsLabelSelector,
+		"app.kubernetes.io/name",
 	)
 
+	var featureStore *feature.Store
+	featureStore = feature.NewStore(logging.FromContext(ctx).Named("feature-config-store"), func(name string, value interface{}) {
+		featureFlags := value.(feature.Flags)
+		if !featureFlags.IsDisabledTransportEncryption() && featureStore != nil {
+			for _, inf := range []injection.InformerInjector{challenge.WithInformer, v1certificate.WithInformer, certificaterequest.WithInformer, clusterissuer.WithInformer, issuer.WithInformer} {
```

Comment:
I think this does not work, when changing the `feature` cfg afterwards. In the `controller.go` of the integrationsink I am getting a `nil` pointer... when accessing the `cmCertificateLister` there.

Reply:
Yes, it doesn't, we could start / stop the informer manually without injection based on the value

Reply:
> we could start / stop the informer manually without injection based on the value

you mean the "cert manager" informers, right? start / stop based on cfg change of the `config-feature` CM?

Any pointer to similar code for this?

Reply:
here we start a "custom" informer using the factory

https://github.com/knative/eventing/blob/b00da5954263042f548c5762bf81343ae734bf2b/pkg/adapter/v2/main.go#L227-L250

the "stop" is based on the passed context and there we can use the usual "context / cancel" pair

Reply:
Thanks

Reply:
We discussed a more dynamic approacj in the past [here](https://github.com/knative/pkg/pull/2976), Serving requires for the [controller pod to be restarted](https://knative.dev/docs/serving/encryption/cluster-local-domain-tls/#enabling-cluster-local-domain-tls). This could be improved but for Serving once you disable encryption there will be downtime anyway afaik. cc @ReToCode if he has anything to add.